genesyscloud/journey_views/genesyscloud_journey_views_proxy.go

```diff
@@ -65,14 +80,78 @@ func getJourneyViewByViewIdFn(_ context.Context, p *journeyViewsProxy, viewId st
 	return p.journeyViewsApi.GetJourneyView(viewId)
 }
 
+func getJourneyViewByNameFn(ctx context.Context, p *journeyViewsProxy, viewName string) (string, *platformclientv2.APIResponse, error) {
+	journeys, resp, err := p.getAllJourneyViews(ctx, viewName)
+	if err != nil {
+		return "", resp, err
+	}
+
+	if journeys == nil || len(*journeys) == 0 {
+		return "", resp, fmt.Errorf("no journey view found with name %s", viewName)
+	}
+
+	for _, journey := range *journeys {
+		if *journey.Name == viewName {
+			fmt.Printf("Retrieved the journey view id %s by name %s", *journey.Id, viewName)
+			return *journey.Id, resp, nil
+		}
+	}
+	return "", resp, fmt.Errorf("unable to find journey view with name %s", viewName)
+}
+
 func createJourneyViewFn(_ context.Context, p *journeyViewsProxy, journeyView *platformclientv2.Journeyview) (*platformclientv2.Journeyview, *platformclientv2.APIResponse, error) {
 	return p.journeyViewsApi.PostJourneyViews(*journeyView)
 }
 
 func updateJourneyViewFn(_ context.Context, p *journeyViewsProxy, viewId string, journeyView *platformclientv2.Journeyview) (*platformclientv2.Journeyview, *platformclientv2.APIResponse, error) {
-	return p.journeyViewsApi.PostJourneyViewVersions(viewId, *journeyView)
+	return p.journeyViewsApi.PostJourneyViewVersions(viewId, *journeyView) //TODO NB
 }
 
 func deleteJourneyViewFn(_ context.Context, p *journeyViewsProxy, viewId string) (*platformclientv2.APIResponse, error) {
 	return p.journeyViewsApi.DeleteJourneyView(viewId)
 }
+
+// GetAllJourneyViewsFn is the implementation for retrieving all journey views in Genesys Cloud
+func getAllJourneyViewsFn(ctx context.Context, p *journeyViewsProxy, name string) (*[]platformclientv2.Journeyview, *platformclientv2.APIResponse, error) {
+	var allJourneys []platformclientv2.Journeyview
+	const pageSize = 100
+
+	journeys, resp, getErr := p.journeyViewsApi.GetJourneyViews(1, pageSize, name, "", "")
+	if getErr != nil {
+		return nil, resp, fmt.Errorf("failed to get first page of journeys: %v", getErr)
+	}
+
+	// Check if the journey view cache is populated with all the data, if it is, return that instead
+	// If the size of the cache is the same as the total number of journeys, the cache is up-to-date
+	if rc.GetCacheSize(p.journeyViewCache) == *journeys.Total && rc.GetCacheSize(p.journeyViewCache) != 0 {
```

Comment:
You don't need this if statement. This logic is specific to routing_queues because the routing_queue cache is shared by other resource.

Reply:
Thanks for this explanation.
I took routing queues as a reference because of the pagination and cache. I simplified this part